li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/producer/LiKafkaProducerBuilder.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2017 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License").  See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.clients.producer;
+
+import java.util.Map;
+
+
+public class LiKafkaProducerBuilder<K, V> {
```

Comment:
Being a public class, it is not clear who should be using this - is it only meant for users of LiFederatedKafkaClients or is it also meant for users of LiKafkaProducer ? Add some documentations to clarify. if we don't want a general usage of this class, we should change it to a non-public scope.

Reply:
This is meant for users of federate clients only. I don't really see users (other than tests) to provide this when constructing federated clients, so I think it makes sense to make it package private.